### PR TITLE
Change implementation of edit and add deadline

### PR DESCRIPTION
**Pull request by evanyan13**
Logic: Change the add and edit Deadline to be executed under `add` and `edit` command

- Modify `AddCommand` & `AddCommandParser` to support adding deadline. 
  But not compulsory to include a deadline field in order to add a new Job.

- Modify `EditCommand` & `EditCommandParser` to support edit deadline.

- Change PREFIX_ROLE from `z/` to `r/`

- Update relevant test cases for AddCommand, AddCommandParser, EditCommand & EditCommandParser.

### CODE REVIEW

File: src/main/java/seedu/application/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -18,12 +20,21 @@ public class AddCommand extends Command {
...
+        + PREFIX_DEADLINE + "DEADLINE \n\n"
+        + "Example: \n"
+        + COMMAND_WORD + " "
+        + PREFIX_COMPANY + "Google"
+        + PREFIX_ROLE + "Software Engineer\n"
+        + "OR\n" + COMMAND_WORD + " "
+        + PREFIX_COMPANY + "Google"
+        + PREFIX_ROLE + "Software Engineer "
+        + PREFIX_STATUS + "pending"
+        + PREFIX_DEADLINE + "Dec 31 2030 1200\n";
```
Comment (migfoo02) [inline src/main/java/seedu/application/logic/commands/AddCommand.java:37]: Good examples to show optional nature of status and deadline.

File: src/main/java/seedu/application/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -24,17 +29,23 @@ public class EditCommand extends Command {
...
+        + "[" + PREFIX_DEADLINE + "DEADLINE] "
+        + "Example: " + COMMAND_WORD + " 1 "
+        + PREFIX_ROLE + "Software Engineer "
+        + PREFIX_COMPANY + "Google"
+        + PREFIX_DEADLINE + "Dec 31 2023 1200";
 
     public static final String MESSAGE_EDIT_JOB_SUCCESS = "Edited Job: %1$s";
-    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided. \n"
+        + "c/ for Company\n"
```
Comment (migfoo02) [inline src/main/java/seedu/application/logic/commands/EditCommand.java:45]: You might want to use `PREFIX_ROLE`, `PREFIX_COMPANY` etc to replace r/, c/ here to keep it consistent.

**Review by migfoo02 — APPROVED** [review]
Overall a good incorporation of deadline into the existing add and edit commands. Just a minor issue with syntax in `EditCommand`

File: src/main/java/seedu/application/logic/commands/EditCommand.java (hunk 2 above)
Comment (evanyan13) [reply src/main/java/seedu/application/logic/commands/EditCommand.java:45]: Yeah good idea good to keep the code consistent if we decide to change the modifier in the future

**Review by JeremyYong128 — APPROVED** [review]
LGTM

**Pull request merged**